src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -613,11 +617,149 @@ public void initialize(HttpRequest request)
+        ArrayList<Permission> adminPerms = new ArrayList<Permission>(editPerms);
+        adminPerms.addAll(new ArrayList<Permission>(Arrays.asList(
+                Computer.CONFIGURE,
+                Computer.DELETE,
+                Hudson.ADMINISTER,
+                Hudson.READ,
+                Run.DELETE,
+                Run.UPDATE,
+                View.CONFIGURE,
+                View.CREATE,
+                View.DELETE,
+                CredentialsProvider.CREATE,
+                CredentialsProvider.UPDATE,
+                CredentialsProvider.DELETE,
+                CredentialsProvider.MANAGE_DOMAINS)));
+        permMap.put("admin", adminPerms);
+        
+        final Credential credential = new Credential(
+                BearerToken.authorizationHeaderAccessMethod())
+                .setAccessToken(getDefaultedClientSecret().getPlainText());
+        HttpRequestFactory requestFactory = transport
+                .createRequestFactory(new HttpRequestInitializer() {
+                    public void initialize(HttpRequest request)
+                            throws IOException {
+                        credential.initialize(request);
+                        request.setParser(new JsonObjectParser(JSON_FACTORY));
+                    }
+                });
+        GenericUrl url = new GenericUrl(getDefaultedServerPrefix() + String.format(CONFIG_MAP_URI, namespace));
+        HttpRequest request = null;
+        ConfigMapResponse response = null;
+        String prefix = "OpenShift Jenkins Login Plugin";
+        try {
+            request = requestFactory.buildGetRequest(url);
+            response = request.execute().parseAs(ConfigMapResponse.class);
+        } catch (IOException e) {
+            LOGGER.info(prefix + " could not find the openshift-jenkins-login-plugin-config config map in namespace " +
+                    namespace + " so the default permission mapping will be used");
+            LOGGER.log(Level.FINE, "getRoleToPermissionMap", e);
+            return permMap;
+        }
+        
+        
+        if (response == null || response.data == null || response.data.size() == 0) {
+            LOGGER.info(prefix + " did not see the openshift-jenkins-login-plugin-config config map in namespace " + 
+                    namespace + " so the default permission mapping will be used");
+            return permMap;
+        }
+        
+        List<Permission> permissionsInSystem = Permission.getAll();
+        for (Entry<String,String> entry : response.data.entrySet()) {
+            String permStr = entry.getKey();
+            
+            if (permStr.trim().equalsIgnoreCase(ConfigMapResponse.ROLE_RANKING)) {
+                String roleList = entry.getValue();
+                if (roleList != null) {
+                    String[] newRoles = roleList.split(",");
+                    if (newRoles != null && newRoles.length > 0) {
+                        roles = new ArrayList<String>(Arrays.asList(newRoles));
```

Comment:
per my comment above, i think we can just build this up by looking at all the roles we see when iterating the mappings.  Once less thing for users to have to specify and potentially screw up.

Reply:
agreed